src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -502,6 +502,25 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 							}
 						}
 						msg = impl.handleApiOther(msg, name, params);
+						break;
+					case pconn:
+						ApiPersistentConnection pconn = impl.getApiPersistentConnection(name);
+						if (pconn != null) {
+							if (!getOptionsParamApi().isDisableKey() && !getOptionsParamApi().isNoKeyForSafeOps()) {
+								if ( ! this.hasValidKey(requestHeader, params)) {
+									throw new ApiException(ApiException.Type.BAD_API_KEY);
+								}
+							}
+							List<String> mandatoryParams = pconn.getMandatoryParamNames();
+							if (mandatoryParams != null) {
+								for (String param : mandatoryParams) {
+									if (!params.has(param) || params.getString(param).length() == 0) {
+										throw new ApiException(ApiException.Type.MISSING_PARAMETER, param);
+									}
+								}
+							}
+						}
+						return impl.handleApiPersistentConnection(msg, httpIn, httpOut, name, params);
```

Comment:
Should the implementation decide if the API request was handled? Maybe the `API` should return always `true`?